components/tidb_query/src/rpn_expr/impl_cast.rs

```diff
@@ -1318,4 +1477,671 @@ mod tests {
             check_warning(&ctx, err_code, log.as_str());
         }
     }
+
+    #[test]
+    fn tes_signed_int_as_signed_real() {
+        test_none_with_nothing(cast_signed_int_as_signed_real);
+
+        let cs: Vec<(i64, f64)> = vec![
+            // (input, expect)
+            (i64::MIN, i64::MIN as f64),
+            (0, 0f64),
+            (i64::MAX, i64::MAX as f64),
+        ];
+
+        for (input, expect) in cs {
+            let r = cast_signed_int_as_signed_real(&Some(input));
+            let r = r.map(|x| x.map(|x| x.into_inner()));
+            let log = make_log(&input, &expect, &r);
+            check_result(Some(&expect), &r, log.as_str());
+        }
+    }
+
+    #[test]
+    fn test_signed_int_as_unsigned_real() {
+        test_none_with_extra(cast_signed_int_as_unsigned_real);
+
+        let cs: Vec<(i64, f64, bool)> = vec![
+            // (input, expect, in_union)
+
+            // TODO: add test case of negative int to unsigned real without in_union
+            // (i64::MIN, i64::MIN as u64 as f64, false),
```

Comment:
Why this case is commented?

Reply:
Because negative number to unsigned real may not right in current cast.